services/dy-jupyter/docker/boot_notebook.sh

```diff
@@ -1,27 +1,55 @@
 #!/bin/bash
-
-set -e
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+IFS=$'\n\t'
 
 # try to pull data from S3
+echo
+echo "trying to restore state..."
 python /docker/state_puller.py ${SIMCORE_NODE_APP_STATE_PATH} --silent
 
 # the notebooks in the folder shall be trusted by default
 # jupyter trust ${SIMCORE_NODE_APP_STATE_PATH}/*
 
 # Trust all notebooks in the notbooks folder
+echo
+echo "trust all notebooks in path..."
 find ${SIMCORE_NODE_APP_STATE_PATH} -name '*.ipynb' | xargs jupyter trust
 
+# prevents notebook to open in separate tab
+cat > ~/.jupyter/custom/custom.js <<EOF
+define(['base/js/namespace'], function(Jupyter){
+    Jupyter._target = '_self';
+});
+EOF
+
+#https://github.com/jupyter/notebook/issues/3130 for delete_to_trash
+cat > jupyter_config.json <<EOF
+{
+    "NotebookApp": {
+        "ip": "0.0.0.0",
+        "port": 8888,
+        "base_url": "${SIMCORE_NODE_BASEPATH}",
+        "extra_static_paths": ["${SIMCORE_NODE_BASEPATH}/static"],
+        "notebook_dir": "${SIMCORE_NODE_APP_STATE_PATH}",
+        "token": "",
+        "quit_button": false,
+        "open_browser": false,
+        "webbrowser_open_new": 0,
+        "nbserver_extensions": {
+            "input_retriever": true,
+            "state_handler": true
+        }
+    },
+    "FileContentsManager": {
+        "post_save_hook": "post_save_hook.export_to_osparc_hook",
+        "delete_to_trash": false
+    },
+    "Session": {
+        "debug": false
+    }
+}
+EOF
+
 # call the notebook with the basic parameters
-start-notebook.sh \
-    --NotebookApp.base_url=${SIMCORE_NODE_BASEPATH} \
-    --NotebookApp.extra_static_paths="['${SIMCORE_NODE_BASEPATH}/static']" \
-    --NotebookApp.notebook_dir=${SIMCORE_NODE_APP_STATE_PATH} \
-    --NotebookApp.token=""  \
-    --NotebookApp.disable_check_xsrf='True' \
-    --NotebookApp.quit_button='False' \
-    --NotebookApp.webbrowser_open_new='0' \
-    --NotebookApp.nbserver_extensions="{'input_retriever':True, 'state_handler':True}" \
-    --FileContentsManager.post_save_hook='post_save_hook.export_to_osparc_hook' \
-    "$@"
-    # --Session.debug='True'
-    # --NotebookApp.token="${NOTEBOOK_TOKEN}" \ this should replace no token and disable_check_xsrf but it currently does not work in platform (reverse proxy?)
+start-notebook.sh --config jupyter_config.json "$@"
```

Comment:
Did you ever find something that consistently does NOT open a new tab? I was playing around with the neuron stuff and it would always open a new tab...

Reply:
well I think I works better now, but I think from time to time it does not work. official documentation  says to edit custom.js which is now done at the last minute to ensure it does not get overwritten by something else.